From 1dc66703908b1182744b4483da99d2b6a37360b1 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Wed, 28 Aug 2024 13:58:30 +0800
Subject: [PATCH 193/277] mtk: mt76: mt7996: fix ldpc setting

The non-AP interfaces would not update conf->vht_ldpc so they never set
STA_CAP_VHT_LDPC even if peer-station support LDPC.
Check conf->vht_ldpc only if it is using AP interface.

Without this patch, station only uses BCC to transmit packet in VHT mode.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/mcu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index 2a4f9a6b..02f05fa4 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -3045,7 +3045,7 @@ mt7996_mcu_sta_rate_ctrl_tlv(struct sk_buff *skb, struct mt7996_dev *dev,
 			cap |= STA_CAP_VHT_TX_STBC;
 		if (link_sta->vht_cap.cap & IEEE80211_VHT_CAP_RXSTBC_1)
 			cap |= STA_CAP_VHT_RX_STBC;
-		if (conf->vht_ldpc &&
+		if ((conf->vif->type != NL80211_IFTYPE_AP || conf->vht_ldpc) &&
 		    (link_sta->vht_cap.cap & IEEE80211_VHT_CAP_RXLDPC))
 			cap |= STA_CAP_VHT_LDPC;
 
-- 
2.45.2

